### PR TITLE
Remove dead codes

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/6233

Problem Summary:

### What is changed and how it works?

As Delta-Tree engine will flush the rows into IStorage layer once we can decode a committed rows from key-values. So there is no need for calling `DBGInvoke __try_flush()`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/fullstack-test/fault-inject/alter-table.test [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-| table_schema | table_name | replica_count | location_labels | available |
-+--------------+------------+---------------+-----------------+-----------+
-| test         | t          |             1 | rack,host,abc   |         1 |
-+--------------+------------+---------------+-----------------+-----------+
->> DBGInvoke __try_flush()
->> DBGInvoke __init_fail_point()
-
-# Process crash when rename column data is done but not for meta.
-# Note that this only affect TxnMergeTree. DeltaTree don't care this since we store
-# column data according to its column-id in TiDB.
```
Comment (JaySon-Huang) [inline tests/fullstack-test/fault-inject/alter-table.test:36]: remove because it is useless now

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by lidezhu — APPROVED** [review]

**Pull request merged**